Production/python/customiseHLT.py

```diff
@@ -0,0 +1,159 @@
+  process.genParticleTable.variables.mass.doc = cms.string('mass')
+
+  return process
+
+
+def customise(process):
+  process.NANOAODSIMoutput = cms.OutputModule("NanoAODOutputModule",
+      compressionAlgorithm = cms.untracked.string('LZMA'),
+      compressionLevel = cms.untracked.int32(9),
+      dataset = cms.untracked.PSet(
+          dataTier = cms.untracked.string('NANOAODSIM'),
+          filterName = cms.untracked.string('')
+      ),
+      fileName = cms.untracked.string('file:nano.root'),
+      outputCommands  = cms.untracked.vstring(
+          'drop *',
+          'keep nanoaodFlatTable_*Table_*_*',
+      )
+  )
+  process.load('PhysicsTools.NanoAOD.nano_cff')
+  from PhysicsTools.NanoAOD.nano_cff import nanoAOD_customizeMC
+
+  #call to customisation function nanoAOD_customizeMC imported from PhysicsTools.NanoAOD.nano_cff
+  process = nanoAOD_customizeMC(process)
+
+  process.nanoAOD_step = cms.Path(process.HLTBeginSequence \
+    + process.HLTL2TauTagNNSequence \
+    + process.HLTGlobalPFTauHPSSequence \
+    + process.HLTHPSDeepTauPFTauSequenceForVBFIsoTau \
+    + process.nanoSequenceMC)
+  process.NANOAODSIMoutput_step = cms.EndPath(process.NANOAODSIMoutput)
+
+  process.tauTable = cms.EDProducer("SimpleCandidateFlatTableProducer",
+    src = cms.InputTag( "hltHpsPFTauProducer" ),
+    cut = cms.string(""),
+    name= cms.string("Tau"),
+    doc = cms.string("HLT taus"),
+    singleton = cms.bool(False), # the number of entries is variable
+    extension = cms.bool(False), # this is the main table
+    variables = cms.PSet(
+      P4Vars,
+      charge = Var("charge", int, doc="electric charge"),
+    )
+  )
+  # cms.EDProducer : an object that produces a new data object
+
+  process.pfCandTable = cms.EDProducer("SimpleCandidateFlatTableProducer",
+    src = cms.InputTag( "hltParticleFlowForTaus" ),
+    cut = cms.string(""),
+    name= cms.string("PFCand"),
+    doc = cms.string("HLT PF candidates for taus"),
+    singleton = cms.bool(False), # the number of entries is variable
+    extension = cms.bool(False), # this is the main table
+    variables = cms.PSet(
+      P4Vars,
+      charge = Var("charge", int, doc="electric charge"),
+
+      energy = Var("energy",float, doc= 'energy'),
+      et = Var("et",float, doc= 'transverse energy'),
+      et2 = Var("et2",float, doc= 'transverse energy squared (use this for cut!)'),
+      massSqr = Var("massSqr",float, doc= 'mass squared'),
+      mt = Var("mt",float, doc= 'transverse mass'),
+      mtSqr = Var("mtSqr",float, doc= 'transverse mass squared'),
+      px = Var("px",float, doc= 'x coordinate of momentum vector'),
+      py = Var("py",float, doc= 'y coordinate of momentum vector'),
+      pz = Var("pz",float, doc= 'z coordinate of momentum vector'),
+      theta = Var("theta",float, doc= 'momentum polar angle'),
+      rapidity = Var("rapidity", float, doc='rapidity'),
+      y = Var("y", float, doc='rapidity'),
```

Comment:
```suggestion
```
all these variables can be re-computed based on (pt, eta, phi, mass) stored above => no need to save them